pkg/hooks/webhook.go

```diff
@@ -66,24 +76,61 @@ func NewWebhookExecutor(
 	if err != nil {
 		return nil, err
 	}
+	return NewWebhookExecutorWithCustomHttpClient(webhook, hookType, client)
+}
+
+func NewWebhookExecutorWithCustomHttpClient(
+	webhook *v1alpha1.Webhook,
+	hookType common.HookType,
+	httpClient HttpClientInterface,
+) (*WebhookExecutor, error) {
+	if webhook == nil {
+		return nil, nil
+	}
+	url, err := webhookURL(webhook)
```

Comment:
is is done in caller method, do not repeat that

Reply:
removed